server/players.py

```diff
@@ -118,31 +113,31 @@ def game_connection(self, value):
     def game_connection(self):
         self._game_connection = lambda: None
 
-    @property
-    def in_game(self):
-        return self.game is not None
-
     def to_dict(self):
         """
         Return a dictionary representing this player object
         :return:
         """
+
         def filter_none(t):
             _, v = t
             return v is not None
-        return dict(filter(filter_none, (
-            ('id', self.id),
-            ('login', self.login),
-            ('global_rating', self.global_rating),
-            ('ladder_rating', self.ladder_rating),
-            ('number_of_games', self.numGames),
-            ('avatar', self.avatar),
-            ('country', self.country),
-            ('clan', self.clan)
-        )))
+
+        return dict(
+            filter(
+                filter_none,
+                (('id', self.id), ('login', self.login),
+                 ('global_rating', self.global_rating),
+                 ('ladder_rating', self.ladder_rating),
+                 ('number_of_games', self.numGames), ('avatar', self.avatar),
+                 ('country', self.country), ('clan', self.clan))
+            )
+        )
```

Comment:
Can we rewrite this code it to dict comprehension?

Reply:
This construction is bad and unreadable.
Also sometimes some tuples are 2x in row, some of them just 1x. it's confusing